index.ts

```diff
@@ -118,6 +201,107 @@ export default class CountTransformer
       stack
     );
 
+    for (const config of this.fields) {
+      const {
+        countFields,
+        field,
+        fields,
+        object,
+        resolverTypeName,
+        resolverFieldName,
+      } = config;
+
+      const localFields = fields.length > 0 ? fields : countFields;
+
+      // Find the table we want to scan
+      const tableDataSource = ctx.dataSources.get(
+        resolverTypeName
+      ) as appsync.DynamoDbDataSource;
+      const table = tableDataSource.ds
+        .dynamoDbConfig as appsync.CfnDataSource.DynamoDBConfigProperty;
+
+      const dataSource = ctx.api.host.getDataSource(
+        `${resolverTypeName.name.value}Table`
+      );
+
+      // Allow the lambda to access this table
+      funcRole.addToPolicy(
+        new iam.PolicyStatement({
+          actions: ["dynamodb:Scan"],
```

Comment:
Should be Query

Reply:
Sorry, which bit in the policy needed to be query?

Reply:
Well unless I'm mistaken, we need the lambda to be able to query the index. We could scan, but that wouldn't be efficient. So add `dynamodb:Query`.